pkg/controller/podautoscaler/metrics/metrics_client_test.go

```diff
@@ -57,9 +57,13 @@ func makeTestServer(t *testing.T, responses map[string]*serverResponse) (*httpte
 	mux := http.NewServeMux()
 
 	mkHandler := func(url string, response serverResponse) *util.FakeHandler {
+		codec, err := testapi.GetCodecForObject(response.obj.(runtime.Object))
```

Comment:
oh, ok, if it's a *typed* v1/v1alpha1 object, I guess that's OK.

On Mon, Sep 21, 2015 at 3:15 PM, Chao Xu <notifications@github.com> wrote:

> In pkg/controller/podautoscaler/metrics/metrics_client_test.go
> <https://github.com/kubernetes/kubernetes/pull/14156#discussion_r40033582>
> :
>
> > @@ -57,9 +57,13 @@ func makeTestServer(t *testing.T, responses map[string]*serverResponse) (*httpte
> >  	mux := http.NewServeMux()
> >
> >  	mkHandler := func(url string, response serverResponse) *util.FakeHandler {
> > +		codec, err := testapi.GetCodecForObject(response.obj.(runtime.Object))
>
> It's because the response.obj in this test sometimes is v1 object, like
> pod, (
> https://github.com/caesarxuchao/kubernetes/blob/move-v1alpha1/pkg/controller/podautoscaler/metrics/metrics_client_test.go#L108
> )
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/14156/files#r40033582>.
>